loading.bs

```diff
@@ -331,6 +385,376 @@ Note: Applying the signed exchange's response here has the effect of letting a
+
+</dl>
+
+<h3 algorithm id="mp-link-type-prefetch">Monkeypatch Link type "prefetch"</h3>
+
+The [=process the linked resource=] given a link element |el|, boolean
+|success|, and [=response=] |response|:
+
+1. If |success| is true, [=fire an event=] named {{HTMLElement/load}} at |el|.
+1. Otherwise, [=fire an event=] named {{HTMLElement/error}} at |el|.
+1. If the |success| is false, then return.
+1. If the |response|'s [=response/came from a signed exchange=] is false, then
+    return.
+1. Let |clonedExchange| be the result of creating a new [=exchange=] with the
+    first item of |response|'s [=response/URL list=] and the result of
+    [=response/cloning=] |response|.
+1. [=map/Set=] |el|'s [=Node/node document=]'s [=Document/prefetched signed
+    exchanges for navigation=][|clonedExchange|'s [=exchange/request URL=]] to
+    |clonedExchange|.
+1. Let |outerLinkHeader| be the result of [=header list/getting=] `` `Link` ``
+    from |response|'s [=response/signed exchange outer header list=].
+1. If |outerLinkHeader| is null, then return.
+1. Let |outerLinks| be the result of [=Parsing a Link Field Value=] from
+    |outerLinkHeader|.
+1. If |outerLinks| is empty, then return.
+1. Let |innerLinkHeader| be the result of [=header list/getting=] `` `Link` ``
+    from |response|'s [=response/header list=].
+1. If |innerLinkHeader| is null, then return.
+1. Let |innerLinks| be the result of [=Parsing a Link Field Value=] from
+    |innerLinkHeader|.
+1. If |innerLinks| is empty, then return.
+1. Let |alternateLinks| be the result of [=getting alternate signed exchange
+    link info=] from |outerLinks|.
+1. Let |allowedSxgLinks| be the result of [=getting allowed signed exchange link
+    info=] from |innerLink|.
+1. For each |innerLink| of |innerLinks|:
+    1. If |innerLink|'s [=Relation Type=] is not 'preload', then continue.
+    1. Let |linkTarget| be |innerLink|'s [=Link Target=].
+    1. Let |asAttribute| be |innerLink|'s [=Target Attribute named=] `"as"`.
+    1. If |asAttribute| is `"image"`, then:
+        1. Let |imagesrcset| be |innerLink|'s [=Target Attribute named=]
+            `"imagesrcset"`.
+        1. Let |imagesizes| be |innerLink|'s [=Target Attribute named=]
+            `"imagesizes"`.
+        1. Create a <{link}> element |linkElement| whose <{link/href}> attribute
+            is |linkTarget|, and <{link/imagesrcset}> attribute is
+            |imagesrcset|, and <{link/imagesizes}> attribute is |imagesizes|.
+        1. Let |selected source| and <var ignore=''>selected pixel density</var>
+            be the URL and pixel density that results from [=selecting an image
+            source=] given |linkElement|, respectively.
+        1. If |selected source| is not null, then set |linkTarget| to the result
+            of [=URL parser|parsing=] |selected source|, with a base URL of
+            |response|'s [=response/URL=].
+    1. For each |allowedSxgLink| of |allowedSxgLinks|:
```

Comment:
This is an O(n^2) algorithm. Is this how we implementing this?
Not a blocker, but might be better to redefine this using a map, as a followup.

Reply:
Blink is using a map AlternateSignedExchangeResourceInfo.EntryMap.

I will do so as a followup.